backend/repository/manager.go

```diff
@@ -321,6 +364,16 @@ func (m *Manager) UpdateAll(r *Repository, clientCfgDir, endpointCfgDir string,
 	if err := m.UpdateEndpoints(r, endpointCfgDir, req.EndpointUpdates); err != nil {
 		return errors.Wrap(err, "failed to update endpoints")
 	}
+
+	if err := m.DeleteEndpoints(r, endpointCfgDir, req.DeleteEndpoints); err != nil {
+		return errors.Wrap(err, "failed to delete endpoints")
+	}
+	if err := m.DeleteClients(r, clientCfgDir, req.DeleteClients); err != nil {
```

Comment:
does this get runs every time when `UpdateAll` is called? I see you created separated path per update thrift/endpoint/clients and some checkings are duplicated. Can we consolidate the delete paths to update path?

Reply:
It does get run every time, but the overhead should be minimal when there aren't any deletions. I originally used the update path for delete, but I switched to a separate one since:

- I can add a flag to ClientConfig and EndpointConfig to indicate deletion, but I'm still going to need a separate path to delete thrifts, since there's no ThriftConfig, just a string right now
- It seems that a "deleteThis" flag doesn't really belong inside a __Config struct. Also, it seems a bit unnecessary to use a __Config struct when only the name and flag are needed for deletion, and this will introduce a new path anyway (e.g. skip the usual input struct validation)

I'm open to other solutions